### PR TITLE
Fix incorrect protocol message

### PR DESCRIPTION
**Pull request by Heath123**
I didn't really make this, so I set it as co-authored
Sets the state to login before sending the message so it gets interpreted properly by the client

### CODE REVIEW

File: src/server/handshake.js [hunk 1]
```diff
@@ -8,6 +8,12 @@ module.exports = function (client, server, { version }) {
     client.serverPort = packet.serverPort
     client.protocolVersion = packet.protocolVersion
 
+    if (packet.nextState === 1) {
+      client.state = states.STATUS
+    } else if (packet.nextState === 2) {
+      client.state = states.LOGIN
+    }
+
```
Comment (DeltaEvo) [inline src/server/handshake.js:16]: You need to assign `client.version` before `client.state` since `version` is used to build the serializer

**Review by DeltaEvo — CHANGES_REQUESTED** [review]

File: src/server/handshake.js (hunk 1 above)
Comment (DeltaEvo) [reply src/server/handshake.js:16]: Or reassign client.state to rebuild the serializer :thinking:

What do you think @rom1504 ?

**Comment by rom1504** [comment]
Does this really work? It seems to me it won't work due to lot of things being done when the state is changed

**Comment by Heath123** [comment]
> Does this really work? It seems to me it won't work due to lot of things being done when the state is changed

I need to update it but I haven't gotten around to it yet

**Comment by rom1504** [comment]
stale, reopen if you want to do it

**Pull request closed without merging**